src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -5467,6 +5469,29 @@ void Unit::RemoveAllAreaTriggers(AreaTriggerRemoveReason reason /*= AreaTriggerR
     }
 }
 
+void Unit::EnterAreaTrigger(AreaTrigger* areaTrigger)
+{
+    m_insideAreaTriggers.push_back(areaTrigger);
+}
+
+void Unit::ExitAreaTrigger(AreaTrigger* areaTrigger)
+{
+    std::erase(m_insideAreaTriggers, areaTrigger);
+}
+
+std::vector<AreaTrigger*> Unit::GetInsideAreaTriggers() const
+{
+    return m_insideAreaTriggers;
+}
+
+void Unit::ExitAllAreaTriggers()
+{
+    for (AreaTrigger* at : m_insideAreaTriggers)
```

Comment:
Will crash in debug mode, HandleUnitExit calls ExitAreaTrigger which can invalidate iterators